server/app/auth/ApiKeyGrants.java

```diff
@@ -0,0 +1,65 @@
+package auth;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+/**
+ * Stores permissions for {@link models.ApiKey}s.
+ *
+ * <p>Permissions are represented as a resource identifier paired with an ability. E.g.
+ * "utility-discount" program with "read" ability. Multiple permissions may be stored for the same
+ * resource. E.g. having both the "read" and "write" permission for a given resource.
+ */
+public class ApiKeyGrants {
+
+  /** Enumerates the abilities an ApiKey may have with respect to a resource */
+  public enum Permission {
+    WRITE,
```

Comment:
Should we remove WRITE for now since we aren't supporting it in v1?

Reply:
I wanted to have more than one permission type so that the unit test can assert correct behavior for handling it.